### DISCUSSION TITLE
Can I have multiple languages running in the same LS?

### DISCUSSION THREAD

**Discussion by luiz290788** [eclipse-langium/langium#976] [Q&A] [answered]
Is it possible to have two different languages running in the same LS Worker in the browser? Or do I have to create multiple workers?

Is it just a matter of registering two language modules in the same shared module?

**Comment by msujew** [comment, marked as answer]
Hey @luiz290788,

we do have an example for multiple languages in one language server over at the [requirements example](https://github.com/langium/langium/tree/main/examples/requirements). Langium doesn't really care whether you're running in a node or browser/webworker context, so that should work as expected.

Note that once you add more than one language to a shared module, the input files need to have the correct file extension to pick the respective language service instance. With just one language, the shared service instance will always return the singleton, independent of the file extension.

**Reply by luiz290788** [reply]
Thanks for the reply!
